### PR TITLE
add pre-operation to add the missing column

### PR DESCRIPTION
**Pull request by kingman**
New column added to the event table, adding update script to add the column to existing event table.

### CODE REVIEW

**Comment by slilichenko** [comment]
I suggest creating a separate backfill script for the new column, perhaps in the new "backfill" or "migration" folder, what the customers will need to run as a one-off script. Mention that script in the comment.

**Review by slilichenko — COMMENTED** [review]
Please take a look at the comments.

File: definitions/digital_analytics_domain/backfill/collected_traffic_source_backfill.sqlx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    ct.collected_traffic_source_id
+  FROM (
+    SELECT
+      event_id,
+      ${ga4.eventUniqueColumns}
+    FROM
+      ${ref(functions.baseSchema("ga4"), "event")}
+    WHERE
+      collected_traffic_source_id IS NULL ) e
+  LEFT JOIN (SELECT * FROM ${ref("ga4_events")}) g_e
```
[the remark this answers is not in the document]
Comment (slilichenko) [reply definitions/digital_analytics_domain/backfill/collected_traffic_source_backfill.sqlx:36]: Perhaps a better way to do it is UPDATE vs MERGE along these lines:
UPDATE events e  SET collected_traffic_source_id = (subselect) WHERE collected_traffic_source_id IS NULL.

Subselect:
select collected_traffic_source_id from ga_events g_e, collected_traffic_source ct WHERE multicolumnequals(e, g_e) and multicolumnequals(g_e, ct). This should return a single row per each event with the missing value

[the remark this answers is not in the document]
Comment (kingman) [reply definitions/digital_analytics_domain/backfill/collected_traffic_source_backfill.sqlx]: Updated according to the comment and tested it. The code now is much cleaner, thank you so much for the suggestion.

**Review by slilichenko — APPROVED** [review]
LGTM

**Pull request merged**